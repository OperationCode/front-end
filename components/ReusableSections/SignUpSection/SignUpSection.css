.content {
  align-self: center;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-around;
  max-width: 50rem;
}
