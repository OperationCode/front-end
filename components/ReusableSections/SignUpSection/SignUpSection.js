import React from 'react';
import Link from 'next/link';
import Section from 'components/_common_/Section/Section';
import Button from 'components/_common_/Button/Button';

const SignUpSection = () => (
  <Section title="New user?" theme="white">
    <p>
      Are you ready to deploy your future? Join Operation Code today and launch your career in
      software development. Once you complete the form below, you&#8217;ll be invited to join our
      team on Slack. Make sure you stop in and say hi!
    </p>

    <Button theme="secondary">
      <Link href="/signup">
        <a>Sign Up</a>
      </Link>
    </Button>
  </Section>
);
export default SignUpSection;
