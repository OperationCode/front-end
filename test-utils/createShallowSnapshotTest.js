import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
