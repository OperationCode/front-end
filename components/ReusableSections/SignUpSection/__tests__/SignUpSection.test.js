import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import SignUpSection from '../SignUpSection';

describe('SignUpSection', () => {
  test('it should render properly with no props', () => {
    createShallowSnapshotTest(<SignUpSection />);
  });
});
