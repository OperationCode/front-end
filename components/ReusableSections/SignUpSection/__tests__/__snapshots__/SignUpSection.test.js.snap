// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpSection it should render properly with no props 1`] = `
<Section
  className=""
  contentClassName=""
  hasHeadingLines={true}
  id=""
  style={Object {}}
  theme="white"
  title="New user?"
>
  <p>
    Are you ready to deploy your future? Join Operation Code today and launch your career in software development. Once you complete the form below, youâ€™ll be invited to join our team on Slack. Make sure you stop in and say hi!
  </p>
  <Button
    analyticsObject={
      Object {
        "action": "Button Selected",
        "category": "Interactions",
      }
    }
    className=""
    datum=""
    disabled={false}
    fullWidth={false}
    onClick={[Function]}
    tabIndex={0}
    theme="secondary"
    type="button"
  >
    <Link
      href="/signup"
    >
      <a>
        Sign Up
      </a>
    </Link>
  </Button>
</Section>
`;
